src/workflows/airqo_etl_utils/storage/cloud_storage.py

```
@@ -0,0 +1,448 @@
+        format: str = "csv",
+    ) -> str:
+        try:
+            with NamedTemporaryFile(delete=False) as temp:
+                if format == "csv":
+                    dataframe.to_csv(temp.name, index=False)
+                elif format == "json":
+                    dataframe.to_json(temp.name, orient="records")
+                elif format == "parquet":
+                    dataframe.to_parquet(temp.name)
+                else:
+                    raise ValueError(f"Unsupported format: {format}")
```

Ruff (0.15.0)

[warning] Abstract `raise` to an inner function

(TRY301)